arch/arm64/core/cortex_r/arm_mpu.c

```diff
@@ -191,6 +252,12 @@ void z_arm64_mm_init(bool is_primary_core)
 		region_init(r_index, &mpu_config.mpu_regions[r_index]);
 	}
 
+	/* DT-defined MPU regions. */
+	if (mpu_configure_regions_from_dt(&static_regions_num) == -EINVAL) {
+		__ASSERT(0, "Failed to allocate MPU regions from DT\n");
+		return;
+	}
+
```

Comment:
Hi Carlo, I saw there is a difference compared to ARM. This should be put below the following line `static_regions_num = mpu_config.num_regions;`

```diff
diff --git a/arch/arm64/core/cortex_r/arm_mpu.c b/arch/arm64/core/cortex_r/arm_mpu.c
index f1486bc445..c0b1724057 100644
--- a/arch/arm64/core/cortex_r/arm_mpu.c
+++ b/arch/arm64/core/cortex_r/arm_mpu.c
@@ -252,15 +252,15 @@ void z_arm64_mm_init(bool is_primary_core)
                region_init(r_index, &mpu_config.mpu_regions[r_index]);
        }

+       /* Update the number of programmed MPU regions. */
+       static_regions_num = mpu_config.num_regions;
+
        /* DT-defined MPU regions. */
        if (mpu_configure_regions_from_dt(&static_regions_num) == -EINVAL) {
                __ASSERT(0, "Failed to allocate MPU regions from DT\n");
                return;
        }

-       /* Update the number of programmed MPU regions. */
-       static_regions_num = mpu_config.num_regions;
-
        arm_core_mpu_enable();

        if (!is_primary_core) {
```

Reply:
well spotted, thanks! I'll fix that after gathering more comments and reviews.